api/lib/getJSON.js

```diff
@@ -45,6 +68,24 @@ const prepVerse = (row, includeMeta = false, liveSearch = 0) => {
     verse.lineNo = row.LineNo;
     verse.updated = row.Updated;
     verse.visraam = JSON.parse(row.Visraam);
+    /* eslint-disable no-param-reassign */
+    if (Array.isArray(verse.visraam)) {
+      const wordCount = JSON.parse(row.WordCount) || [0];
+      let totalWordCount = 0;
+      const accumulator = {};
+      verse.visraam.forEach((line, i) => {
+        Object.keys(line).forEach(vtype => {
+          if (!accumulator[vtype]) accumulator[vtype] = [];
+          line[vtype].forEach(v => {
+            if (i > 0) v.p = parseInt(v.p, 10) + totalWordCount;
+            accumulator[vtype].push(v);
+          });
+        });
+        totalWordCount += wordCount[i];
+      });
+      verse.visraam = accumulator;
```

Comment:
Don't really like this mutation of parameters, would rather just have a return value